src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1,40 +1,934 @@
+    GfxPrint_Printf(printer, "OPT=%d", this->opt);
+}
+
+// clang-format off
+static const char* sLoadingMessages[] = {
+    // "Please wait a minute",
+    "\x8Dｼﾊﾞﾗｸｵﾏﾁｸﾀﾞｻｲ",
+    // "Hold on a sec",
+    "\x8Dﾁｮｯﾄ ﾏｯﾃﾈ",
+    // "Wait a moment",
+    "\x8Cｳｪｲﾄ ｱ ﾓｰﾒﾝﾄ",
+    // "Loading",
+    "\x8Cﾛｰﾄﾞ\x8Dﾁｭｳ",
+    // "Now working",
+    "\x8Dﾅｳ ﾜｰｷﾝｸﾞ",
+    // "Now creating",
+    "\x8Dｲﾏ ﾂｸｯﾃﾏｽ",
+    // "It's not broken",
+    "\x8Dｺｼｮｳｼﾞｬﾅｲﾖ",
+    // "Coffee Break",
+    "\x8Cｺｰﾋｰ ﾌﾞﾚｲｸ",
+    // "Please set B side",
+    "\x8C" "Bﾒﾝｦｾｯﾄｼﾃｸﾀﾞｻｲ",
+    // "Be patient, now",
+    "\x8Dｼﾞｯﾄ\x8Cｶﾞﾏﾝ\x8Dﾉ\x8Cｺ\x8Dﾃﾞｱｯﾀ",
+    // "Please wait just a minute",
+    "\x8Dｲﾏｼﾊﾞﾗｸｵﾏﾁｸﾀﾞｻｲ",
+    // "Don't worry, don't worry. Take a break, take a break",
+    "\x8Dｱﾜﾃﾅｲｱﾜﾃﾅｲ｡ﾋﾄﾔｽﾐﾋﾄﾔｽﾐ｡",
+};
+// clang-format on
+
+void Select_PrintLoadingMessage(SelectContext* this, GfxPrint* printer) {
+    s32 index;
+
+    GfxPrint_SetPos(printer, 10, 15);
+    GfxPrint_SetColor(printer, 255, 255, 255, 255);
+
+    index = Rand_ZeroOne() * ARRAY_COUNT(sLoadingMessages);
+    GfxPrint_Printf(printer, "%s", sLoadingMessages[index]);
+}
+
+// clang-format off
+// Second column is unused
+static const char* sFormLabel[][2] = {
+    // "17 (Adult)" // 17 (Daitetsujin)
+    {"\x8D""17(ｵﾄﾅ)",    "\x8D""17(ﾀﾞｲﾃﾂｼﾞﾝ)"},
+    // "30 (Goron)" // 30 (Ice Cream -1)
+    {"\x8C""30(ｺﾞﾛﾝ)",   "\x8C""30(ｱｲｽｸﾘｰﾑ-1)"},
+    // "78 (Zora)" // 78 (Carmen +1)
+    {"\x8C""78(ｿﾞｰﾗ)",   "\x8C""78(ｶﾙﾒﾝ+1)"},
+    // "12 (Deku)" // 12 (Majestic)
+    {"\x8C""12(ﾃﾞｸﾅｯﾂ)", "\x8C""12(ﾏｼﾞｪｽﾃｨｯｸ)"},
+    // "5 (Child)" // 5 (NTT Kodomo)
+    {"\x8D""5(ｺﾄﾞﾓ)",    "\x8C""5(NTTｺﾄﾞﾓ)"},
+};
+// clang-format on
+
+void Select_PrintAgeSetting(SelectContext* this, GfxPrint* printer, s32 playerForm) {
+    s32 pad;
+    const char* age;
+    const char*(*ageTemp)[2];
+    const char*(*phi_v1)[2];
+    s32 phi_v0;
+
+    if ((playerForm >= PLAYER_FORM_FIERCE_DEITY) && (playerForm < PLAYER_FORM_MAX)) {
+        ageTemp = &sFormLabel[playerForm];
+        phi_v1 = ageTemp + 1;
+
+        // This loop doesn't do anything
+        for (phi_v0 = 8; phi_v0 >= 0; phi_v0 -= 8) {
+            if (phi_v1 != 0) {
+                break;
+            }
+            phi_v1--;
+        }
+        if (phi_v1) {}
+
+        age = (*ageTemp)[0];
+    } else {
+        age = NULL;
+    }
+
+    GfxPrint_SetPos(printer, 4, 26);
+    GfxPrint_SetColor(printer, 255, 255, 55, 255);
+    if (age != NULL) {
+        GfxPrint_Printf(printer, "Age:%s", age);
+    } else {
+        // clang-format off
+        GfxPrint_Printf(printer, "Age:" "???" "(%d)", playerForm);
+        // clang-format on
+    }
+}
+
+void Select_PrintCutsceneSetting(SelectContext* this, GfxPrint* printer, u16 csIndex) {
+    const char* stage;
+    const char* day;
+
+    GfxPrint_SetPos(printer, 4, 25);
+    GfxPrint_SetColor(printer, 255, 255, 55, 255);
+
+    // "-jara" used in these strings is a Kokiri speech quirk word
+    switch (csIndex) {
```

Comment:
This `csIndex`, along with `gSaveContext.cutscene` is interesting to me, how does this cutscene index compare to the cutscene index used in `eventMgr` which deals with scene cutscenes and global cutscenes. The other `csIndex` deals with a range of `0-7F`, so I wonder if they're different. If they're the same, that makes things easier. But if they're different, which I suspect so, we should discuss in discord how to differentiate between them. But this is also outside the scope of this PR, just something we need to look into.